### PR TITLE
Update from node 16 to node 20 for Github actions and builds

### PR DESCRIPTION
**Pull request by lucasnetau**
Node 16 in GitHub actions is deprecated. Update actions to use node 20 plus dependancy updates to bring it up to node 20 compatibility. In doing do a lot of old unmaintained packages have been removed. Babel, ESlint and Jest updated to recent versions.

There are a few remaining packages that will need to wait for a future update from dependancies (Babel to v8, Jest JS Env, Semantic release)

### CODE REVIEW

**Comment by lucasnetau** [comment]
@kevinchappell my local versions are working fine but another set of eyes on this would be useful.

**Review by kevinchappell — APPROVED** [review]
In the near future I'd like to upgrade all the way to v22 to take advantage of some of the newer built-in features for testing and task running. The languages repo was upgraded to node 22 and I was able to remove a few dependencies and eslint and prettier were replaced with biomejs.

**Comment by lucasnetau** [comment]
I'll rebase package-lock.json and force-push again. This PR will need to be applied before any others to ensure no merge conflicts after semantic release updates the file version

**Pull request merged**
